src/transport/MessageHeader.cpp

```diff
@@ -251,13 +261,15 @@ CHIP_ERROR PacketHeader::Encode(uint8_t * data, size_t size, size_t * encode_siz
     LittleEndian::Write16(p, mPayloadLength);
 
     // Written data size provided to caller on success
-    *encode_size = p - data;
+    VerifyOrExit(p - data == EncodeSizeBytes(), err = CHIP_ERROR_INTERNAL);
+    static_assert(std::is_same<decltype(EncodeSizeBytes()), uint16_t>::value, "Unexpected EncodeSizeBytes() return type");
```

Comment:
nit: This seems overfitting. I would drop it. Same for the items below.

Headers are already uint16_t. Not sure why we would need a static assert that the header defines it in a specific way.

Reply:
So just remove the static asserts?  I can do that....  As for why I had this:

> Headers are already uint16_t.

Well, right, this was mean to catch someone trying to change that and avoid bugs if that happens.

> Not sure why we would need a static assert that the header defines it in a specific way.

The combination of that assert and the dynamic check that `p - data == EncodeSizeBytes()` is what guarantees that casting `p - data` to `uint16_t` is safe.

We could more explicitly test that `p - data` fits in `uint16_t` if you prefer; this just seemed like a way to avoid a dynamic check.

Reply:
Let's remove the decltype static asserts. 
Eventually we can verify or exit that `p-data <= UINT16_MAX` to be really safe.

Reply:
Per offline conversation, will remove the static_assert.